commit 9663141274e01592a281a7f2df5d7a31a1dac8bf
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Jan 20 21:53:14 2023 +0800

    liblzma: Set documentation on all reserved fields to private.
    
    This prevents the reserved fields from being part of the generated
    Doxygen documentation.

diff --git a/src/liblzma/api/lzma/base.h b/src/liblzma/api/lzma/base.h
index 5e44ad5e..e43bda93 100644
--- a/src/liblzma/api/lzma/base.h
+++ b/src/liblzma/api/lzma/base.h
@@ -547,9 +547,17 @@ typedef struct {
 	 * you should not touch these, because the names of these variables
 	 * may change.
 	 */
+
+	/** \private     Reserved field. */
 	void *reserved_ptr1;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr2;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr3;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr4;
 
 	/**
@@ -564,10 +572,19 @@ typedef struct {
 	 */
 	uint64_t seek_pos;
 
+	/** \private     Reserved field. */
 	uint64_t reserved_int2;
+
+	/** \private     Reserved field. */
 	size_t reserved_int3;
+
+	/** \private     Reserved field. */
 	size_t reserved_int4;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum1;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum2;
 
 } lzma_stream;
diff --git a/src/liblzma/api/lzma/block.h b/src/liblzma/api/lzma/block.h
index 1cf928e6..58e6b37f 100644
--- a/src/liblzma/api/lzma/block.h
+++ b/src/liblzma/api/lzma/block.h
@@ -224,20 +224,50 @@ typedef struct {
 	 * with the currently supported options, so it is safe to leave these
 	 * uninitialized.
 	 */
+
+	/** \private     Reserved field. */
 	void *reserved_ptr1;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr2;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr3;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int1;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int2;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int3;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int4;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int5;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int6;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int7;
+
+	/** \private     Reserved field. */
 	lzma_vli reserved_int8;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum1;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum2;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum3;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum4;
 
 	/**
@@ -261,12 +291,25 @@ typedef struct {
 	 */
 	lzma_bool ignore_check;
 
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool2;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool3;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool4;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool5;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool6;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool7;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool8;
 
 } lzma_block;
diff --git a/src/liblzma/api/lzma/container.h b/src/liblzma/api/lzma/container.h
index 4e732e37..fcbe5734 100644
--- a/src/liblzma/api/lzma/container.h
+++ b/src/liblzma/api/lzma/container.h
@@ -172,12 +172,25 @@ typedef struct {
 	 * with the currently supported options, so it is safe to leave these
 	 * uninitialized.
 	 */
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum1;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum2;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum3;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int1;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int2;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int3;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int4;
 
 	/**
@@ -223,11 +236,22 @@ typedef struct {
 	 */
 	uint64_t memlimit_stop;
 
+	/** \private     Reserved field. */
 	uint64_t reserved_int7;
+
+	/** \private     Reserved field. */
 	uint64_t reserved_int8;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr1;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr2;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr3;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr4;
 
 } lzma_mt;
diff --git a/src/liblzma/api/lzma/delta.h b/src/liblzma/api/lzma/delta.h
index 84234b6d..775647e1 100644
--- a/src/liblzma/api/lzma/delta.h
+++ b/src/liblzma/api/lzma/delta.h
@@ -68,11 +68,23 @@ typedef struct {
 	 * when type is LZMA_DELTA_TYPE_BYTE, so it is safe to leave these
 	 * uninitialized.
 	 */
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int1;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int2;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int3;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int4;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr1;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr2;
 
 } lzma_options_delta;
diff --git a/src/liblzma/api/lzma/index.h b/src/liblzma/api/lzma/index.h
index 93dba1a5..c7be9772 100644
--- a/src/liblzma/api/lzma/index.h
+++ b/src/liblzma/api/lzma/index.h
@@ -51,8 +51,13 @@ typedef struct {
 		 */
 		const lzma_stream_flags *flags;
 
+		/** \private     Reserved field. */
 		const void *reserved_ptr1;
+
+		/** \private     Reserved field. */
 		const void *reserved_ptr2;
+
+		/** \private     Reserved field. */
 		const void *reserved_ptr3;
 
 		/**
@@ -108,9 +113,17 @@ typedef struct {
 		 */
 		lzma_vli padding;
 
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli1;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli2;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli3;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli4;
 	} stream;
 
@@ -197,14 +210,28 @@ typedef struct {
 		 */
 		lzma_vli total_size;
 
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli1;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli2;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli3;
+
+		/** \private     Reserved field. */
 		lzma_vli reserved_vli4;
 
+		/** \private     Reserved field. */
 		const void *reserved_ptr1;
+
+		/** \private     Reserved field. */
 		const void *reserved_ptr2;
+
+		/** \private     Reserved field. */
 		const void *reserved_ptr3;
+
+		/** \private     Reserved field. */
 		const void *reserved_ptr4;
 	} block;
 
diff --git a/src/liblzma/api/lzma/lzma12.h b/src/liblzma/api/lzma/lzma12.h
index 1b1ba2b3..cd520520 100644
--- a/src/liblzma/api/lzma/lzma12.h
+++ b/src/liblzma/api/lzma/lzma12.h
@@ -481,16 +481,38 @@ typedef struct {
 	 * with the currently supported options, so it is safe to leave these
 	 * uninitialized.
 	 */
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int4;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int5;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int6;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int7;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int8;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum1;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum2;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum3;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum4;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr1;
+
+	/** \private     Reserved field. */
 	void *reserved_ptr2;
 
 } lzma_options_lzma;
diff --git a/src/liblzma/api/lzma/stream_flags.h b/src/liblzma/api/lzma/stream_flags.h
index c8c7b22a..6eb19c33 100644
--- a/src/liblzma/api/lzma/stream_flags.h
+++ b/src/liblzma/api/lzma/stream_flags.h
@@ -88,19 +88,47 @@ typedef struct {
 	 * is just two bytes plus Backward Size of four bytes. But it's
 	 * nice to have the proper types when they are needed.)
 	 */
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum1;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum2;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum3;
+
+	/** \private     Reserved field. */
 	lzma_reserved_enum reserved_enum4;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool1;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool2;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool3;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool4;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool5;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool6;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool7;
+
+	/** \private     Reserved field. */
 	lzma_bool reserved_bool8;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int1;
+
+	/** \private     Reserved field. */
 	uint32_t reserved_int2;
 
 } lzma_stream_flags;